tools/hive-bigquery/hive_to_bigquery.py

```diff
@@ -0,0 +1,257 @@
+                                        gcs_component, hive_table_model,
+                                        bq_table_model,
+                                        PropertiesReader.get('gcs_bucket_name'),
+                                        tracking_data)
+
+            # Updates BigQuery job status and wait for all the jobs to finish
+            bq_component.update_bq_job_status(mysql_component, gcs_component,
+                                              hive_table_model, bq_table_model,
+                                              PropertiesReader.get(
+                                                  'gcs_bucket_name'))
+
+        else:
+            print_and_log(
+                "Tracking table already exists. Continuing from the previous "
+                "iteration...")
+
+            # Copies the pending files from the previous run to GCS, loads them
+            # to BigQuery and updates the BigQuery load job status
+            gcs_component.stage_to_gcs(mysql_component, bq_component,
+                                       hive_table_model, bq_table_model,
+                                       PropertiesReader.get('gcs_bucket_name'))
+            bq_component.load_gcs_to_bq(mysql_component, hive_table_model,
+                                        bq_table_model)
+            bq_component.update_bq_job_status(mysql_component, gcs_component,
+                                              hive_table_model, bq_table_model,
+                                              PropertiesReader.get(
+                                                  'gcs_bucket_name'))
+
+        # Checks for new data in the Hive table
+        tracking_data = hive_component.check_inc_data(
+            mysql_component, bq_component, gcs_component, hive_table_model,
+            bq_table_model, PropertiesReader.get('gcs_bucket_name'))
+
+        if tracking_data:
+            # Migrates data to BigQuery and updates job status in the tracking
+            # table
+            hive_component.migrate_data(mysql_component, bq_component,
+                                        gcs_component, hive_table_model,
+                                        bq_table_model,
+                                        PropertiesReader.get('gcs_bucket_name'),
+                                        tracking_data)
+            bq_component.update_bq_job_status(mysql_component, gcs_component,
+                                              hive_table_model, bq_table_model,
+                                              PropertiesReader.get(
+                                                  'gcs_bucket_name'))
+
+        # Compares the number of rows in BigQuery and Hive tables and creates
+        # metrics table if there is a match
+        compare_rows(bq_component, hive_component, gcs_component,
+                     hive_table_model, bq_table_model)
+
+    except Exception as error:
+        logger.exception(error)

```

Comment:
You need to fix your error handling habits. using this print-log-exit is ugly, untraceable and not extensible.

refactor this to be: 
```
        rollback(mysql_component, hive_table_model)
        logger.error(error)
```